lib/loader/serialize/serial_instruction.cpp

```diff
@@ -146,19 +146,79 @@ Serializer::serializeInstruction(const AST::Instruction &Instr,
 
   // Reference Instructions.
   case OpCode::Ref__null:
+  case OpCode::Ref__test:
+  case OpCode::Ref__cast:
+  case OpCode::Ref__test_null:
+  case OpCode::Ref__cast_null:
```

Comment:
This is incorrect.
According to the [binrary format](https://webassembly.github.io/gc/core/binary/instructions.html#reference-instructions) of `ref.test` and `ref.cast` instructions, the immediates are the `heaptype`, not `reftype`.
You can refer to our [implementation here](https://github.com/WasmEdge/WasmEdge/blob/master/lib/loader/ast/instruction.cpp#L520-L538).
We loaded these instructions with their `heaptype` immediates, and set into the `valtype` class.
Therefore, for the serialization, you should output the [`heaptype`](https://webassembly.github.io/gc/core/binary/types.html#binary-heaptype) without the `ref` or `ref.null` decoration.